runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -235,10 +241,32 @@ private void executeMonitorHeld(Runnable work, long workBytes) {
   }
 
   private void decrementCounters(long workBytes) {
-    monitor.enter();
-    --elementsOutstanding;
-    bytesOutstanding -= workBytes;
-    monitor.leave();
+    // All threads queue decrements and one thread grabs the monitor and updates
+    // counters. We do this to reduce contention on monitor which is locked by
+    // GetWork thread
+    decrementQueue.add(workBytes);
```

Comment:
think we get away with an atomic integer. 

static long ITEM_COUNT = 1L << 46;
atomicCounters.getAndAdd(ITEM_COUNT & workBytes);
synchronized (...) {
  long flushedCounter = atomicCounters.getAndSet(0);
  long items = flushedCounter >> 50;
  long bytes = flushedCounter & (ITEM_COUNT - 1);
  // could check for overflow
  monitor...
}

Reply:
I think keeping the queue is simpler than packing both values into an atomic + overflow checking. The queue should contain only one element per harness thread. Let's keep the queue.